build-tools/create-dotnet-msi/Microsoft.NET.Workload.Android.wix.in

```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Wix xmlns="http://schemas.microsoft.com/wix/2006/wi">
+
+  <Product Name="Microsoft.NET.Workload.Android"
+      Id="*"
+      Language="1033"
+      Version="@MSIVERSION@"
+      Manufacturer="Microsoft"
+      UpgradeCode="86a83cc9-6b75-489f-be10-27988df907c6">
+    <Package InstallerVersion="200" Compressed="yes" InstallScope="perMachine" Platform="x64" />
+    <MajorUpgrade DowngradeErrorMessage="A newer version of [ProductName] is already installed." />
```

Comment:
`[ProductName]`?  Is that replaced by WiX?

Reply:
`[ProductName]` should get replaced by the `Name` we have here:
```xml
<Product Name="Microsoft.NET.Workload.Android"
```
This message came from the File -> New template. I'm not sure how to get it to display this error message, though.

On reinstall I get this popup:

![image](https://user-images.githubusercontent.com/840039/97042396-96f52e80-1536-11eb-9c52-d1edb484fa1e.png)

Reply:
@jonathanpeppers Right now you're using `AndroidMSIVersion` for the MSI version but this will be problematic for MSI upgrades the way you have it right now since Windows installer only takes the first three parts of a version number into account (https://docs.microsoft.com/en-us/windows/win32/msi/upgrade-table):
> Note that Windows Installer uses only the first three fields of the product version. If you include a fourth field in your product version, the installer ignores the fourth field.

If you want to support upgrading to an .msi where only the fourth field changed then you need to set `AllowSameVersionUpgrades="yes"` on the `<MajorUpgrade>` node in the Wix source: https://wixtoolset.org/documentation/manual/v3/xsd/wix/majorupgrade.html

Reply:
And to answer the initial question, yes this placeholder gets replaced by the MSI product name and shows in an error message when you try to downgrade from a later version to an earlier version.

Reply:
@akoeplinger adding `AllowSameVersionUpgrades="yes"` causes a warning during the build now:
```
warning LGHT1076: ICE61: This product should remove only older versions of itself. The Maximum version is not less than the current product. (11.0.100.205 11.0.100.205)
```
Is there some other "maximum version" I should also set?

It's kind of hard to find info on WIX, but some are saying you will always get this warning?

https://stackoverflow.com/questions/37941133/ice61-this-product-should-remove-only-older-versions-of-itself

Reply:
After more reading, I think `AllowSameVersionUpgrades="yes"` and suppressing the ICE61 warning is the way to go.

Reply:
yeah that's we do in Mono, we suppress ICE61: https://github.com/mono/mono/blob/c5b88ec4f323f2bdb7c7d0a595ece28dae66579c/packaging/Windows/resources/MonoForWindows.wixproj#L12